### PR TITLE
Add godoc-style docstrings to Cmd... methods

### LINKED ISSUE #11596: Missing documentation on all API methods
**Problem**
 
Document DockerCli `Cmd...` method
 
**Solution**
 
Write godoc-style docstrings for each `Cmd...` method(which cli command it represents, maybe something about parameters) in `api/client/commands.go`.

### PR DESCRIPTION
**Pull request by peggyl**
Adds docstrings for all docker Cmd... methods based on the existing help messages for the cli subcommands. For optional flags, this defers to the docker cli so that the godoc remains concise.

Verified with godoc on localhost.

Closes #11596.

### CODE REVIEW

**Comment by LK4D4** [comment]
@peggyl Omg, so awesome. Feel free to find more packages for document and create issues and PRs for them.

**Comment by icecrime** [comment]
OMG that's so good to have, thanks! Closes #11596.

**Comment by jfrazelle** [comment]
THIS IS AWESOME

File: api/client/help.go [hunk 1]
```diff
@@ -7,6 +7,11 @@ import (
 	flag "github.com/docker/docker/pkg/mflag"
 )
 
+// CmdHelp displays information on a Docker command.
+//
+//If more than one command is specified, information is only shown for the first command.
```
Comment (icecrime) [inline api/client/help.go]: There are a few missing whitespace at the beginning of lines like that one, but not a blocker for me.

File: api/client/commit.go [hunk 2]
```diff
@@ -14,6 +14,9 @@ import (
 	"github.com/docker/docker/utils"
 )
 
+// CmdAttach attaches to a running container.
+//
+// Usage: docker attach [OPTIONS] CONTAINER
 func (cli *DockerCli) CmdCommit(args ...string) error {
```
Comment (tiborvass) [inline api/client/commit.go]: Wrong copy/pasta? ;)
It's the commit command but with attach docs

**Comment by LK4D4** [comment]
LGTM

**Comment by crosbymichael** [comment]
LGTM

@peggyl thanks!

**Pull request merged**